### PR TITLE
fix spine cached color format

### PR DESCRIPTION
**Pull request by zhakesi**

### CODE REVIEW

File: native/cocos/editor-support/spine-creator-support/SkeletonCache.cpp [hunk 1]
```diff
@@ -317,15 +319,15 @@ void SkeletonCache::renderAnimationFrame(AnimationData *animationData) {
             _clipper->clipEnd(*slot);
             continue;
         }
-
+        spine::Color slotColor = slot->getColor();
```
Comment (minggo) [inline native/cocos/editor-support/spine-creator-support/SkeletonCache.cpp]: It is better to use const reference.

File: native/cocos/editor-support/spine-creator-support/SkeletonCache.cpp [hunk 2]
```diff
@@ -394,32 +388,42 @@ void SkeletonCache::renderAnimationFrame(AnimationData *animationData) {
...
+            darkColor.a = 0;
         } else {
             darkColor.r = 0;
             darkColor.g = 0;
             darkColor.b = 0;
+            darkColor.a = 0;
         }
-        darkColor.a = 0;
+
+        finalColor.r = static_cast<uint8_t>(round(color.r));
```
Comment (minggo) [inline native/cocos/editor-support/spine-creator-support/SkeletonCache.cpp]: Use `std::round()` instead?

File: native/cocos/editor-support/spine-creator-support/SkeletonCache.cpp (hunk 1 above)
Comment (zhakesi) [reply native/cocos/editor-support/spine-creator-support/SkeletonCache.cpp]: done

File: native/cocos/editor-support/spine-creator-support/SkeletonCache.cpp (hunk 2 above)
Comment (zhakesi) [reply native/cocos/editor-support/spine-creator-support/SkeletonCache.cpp]: done

**Review by SantyWang — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**